From 59f8ca68d180be3b527098ea386fb54f8c74c501 Mon Sep 17 00:00:00 2001
From: Jiefeng Ou <oujiefeng@h-partners.com>
Date: Thu, 21 Apr 2022 21:49:57 +0800
Subject: [PATCH] PCI: fix kabi change in struct pci_dev
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 59f8ca68d180be3b527098ea386fb54f8c74c501
Modified-by-SEL: No


driver inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I51U4T

--------------------------------------------------------------------------

Fix kabi change in struct pci_dev since the following patches:
- commit 8eb7b6ca203f ("PCI/ERR: Cache RCEC EA Capability offset in
  pci_init_capabilities()")
- commit 1345ecf47242 ("PCI/ERR: Add pcie_link_rcec() to associate RCiEPs")

Signed-off-by: Jiefeng Ou <oujiefeng@h-partners.com>
Reviewed-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Jay Fang <f.fangjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/pci.h | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/include/linux/pci.h b/include/linux/pci.h
index 7d4e6c223970..aee7a7563fc2 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -330,10 +330,6 @@ struct pci_dev {
 #ifdef CONFIG_PCIEAER
 	u16		aer_cap;	/* AER capability offset */
 	struct aer_stats *aer_stats;	/* AER stats for this device */
-#endif
-#ifdef CONFIG_PCIEPORTBUS
-	struct rcec_ea	*rcec_ea;	/* RCEC cached endpoint association */
-	struct pci_dev  *rcec;          /* Associated RCEC device */
 #endif
 	u8		pcie_cap;	/* PCIe capability offset */
 	u8		msi_cap;	/* MSI capability offset */
@@ -516,8 +512,18 @@ struct pci_dev {
 	unsigned long   slot_being_removed_rescanned;
 	struct pci_dev  *rpdev; /* root port pci_dev */
 
+#ifndef __GENKSYMS__
+#ifdef CONFIG_PCIEPORTBUS
+	struct rcec_ea	*rcec_ea;	/* RCEC cached endpoint association */
+	struct pci_dev  *rcec;          /* Associated RCEC device */
+#else
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+#endif
+#else
 	KABI_RESERVE(1)
 	KABI_RESERVE(2)
+#endif
 	KABI_RESERVE(3)
 	KABI_RESERVE(4)
 	KABI_RESERVE(5)
-- 
2.34.1

